src/realm/obj.cpp

```diff
@@ -761,10 +762,102 @@ void out_mixed(std::ostream& out, const Mixed& val)
     }
 }
 
+void out_mixed_xjson(std::ostream& out, const Mixed& val)
+{
+    if (val.is_null()) {
+        out << "null";
+        return;
+    }
+    switch (val.get_type()) {
+        case type_Int:
+            out << "{\"$numberLong\": \"";
+            out << val.get<Int>();
+            out << "\"}";
+            break;
+        case type_Bool:
+            out << (val.get<bool>() ? "true" : "false");
+            break;
+        case type_Float:
+            out << "{\"$numberDouble\": \"";
+            out_floats<float>(out, val.get<float>());
+            out << "\"}";
+            break;
+        case type_Double:
+            out << "{\"$numberDouble\": \"";
+            out_floats<double>(out, val.get<double>());
+            out << "\"}";
+            break;
+        case type_String: {
+            out << "\"";
```

Comment:
The following could be factored out in a out_string() function and reused in `out_mixed_json`. This will reduce the duplicate code. The same goes for binary.